BackendIntegrationTests/BackendIntegrationTests.swift

```diff
@@ -48,6 +48,7 @@ class BackendIntegrationSK1Tests: XCTestCase {
         }
 
         testSession = try SKTestSession(configurationFileNamed: Constants.storeKitConfigFileName)
+        testSession.resetToDefaultState()
```

Comment:
Even though we're creating a new session for every test, after the new test sets `askToBuyEnabled` to `true`, the following tests inherit this, so internally the test session seems to be shared. This ensures all tests start from a clean state.